### DISCUSSION TITLE
Microphone support

### DISCUSSION THREAD

**Discussion by jadasse** [86Box/86Box#3872] [Ideas]
**Is your feature request related to a problem? Please describe.**
Whenever I feel like recording something in some of my VMs, but when I stop recording, I am playing back nothing.

**Describe the solution you would like**
I think the best way to solve this is for you guys to add sound-in device support (e.g., a microphone input)

**Additional context**
By the way, my microphone is built-in.

**Comment by Torinde** [comment]
joncampbell123/dosbox-x/issues/1824

There are [6 DOS, 30 Win3.x games and ~150 Win9x/XP games](https://www.mobygames.com/game/attribute:719/platform:dos/platform:win3x/sort:date/page:1/) supporting microphone input.

[DOSbox-X](https://github.com/joncampbell123/dosbox-x/issues/3999#issuecomment-1426873626):
> the framework is now there in the Sound Blaster emulation for recording audio in general, though at this time, only generated fake source are supported. They are: silence, hiss, and a 1KHz test tone. If you record audio in the guest the recording should contain that generated audio.

[Some further TODO items (and lists of games and software)](https://github.com/joncampbell123/dosbox-x/issues/1824#issuecomment-687309564):
> default to the fake silence source until the user confirms with the interface they do in fact intend for the DOS program to record them
Faking input from a WAV file

**Reply by Torinde** [reply]
> Microphone is required for some interactive games like Fin Fin.

- #4675

**Reply by Torinde** [reply]
[MAME 0.278](https://www.mamedev.org/?p=552):
- Native WASAPI support on Windows and PipeWire support on Linux.
- Support for sound input for emulated systems that have microphones or other audio capture hardware.
- Support for multi-channel input and output.

**Reply by jadasse** [reply]
> [joncampbell123/dosbox-x/issues/1824](https://github.com/joncampbell123/dosbox-x/issues/1824)
> 
> There are [6 DOS, 30 Win3.x games and ~150 Win9x/XP games](https://www.mobygames.com/game/attribute:719/platform:dos/platform:win3x/sort:date/page:1/) supporting microphone input.
> 
> [DOSbox-X](https://github.com/joncampbell123/dosbox-x/issues/3999#issuecomment-1426873626):
> 
> > the framework is now there in the Sound Blaster emulation for recording audio in general, though at this time, only generated fake source are supported. They are: silence, hiss, and a 1KHz test tone. If you record audio in the guest the recording should contain that generated audio.
> 
> [Some further TODO items (and lists of games and software)](https://github.com/joncampbell123/dosbox-x/issues/1824#issuecomment-687309564):
> 
> > default to the fake silence source until the user confirms with the interface they do in fact intend for the DOS program to record them
> > Faking input from a WAV file

DOSBox-X seems to have finally implemented microphone

**Comment by Torinde** [comment]
Related:
- #4063

**Comment by tokariu** [comment]
I wasn't aware that there is no way to record audio in 86box yet until I tried to. therefor I want to highlight this request, because it makes a lot of sound and sampling tools useless without audio input.

**Comment by Hornet512** [comment]
i wanted this too, i think it would be a nice feature, since VirtualBox can do it, but 86box has better emulation accuracy.

**Comment by DFXThomas** [comment]
Oh, yes. Please make this work. I'd love to use Windows 3.1's audio recorder again!

**Comment by DFXThomas** [comment]
I am still hoping / begging for this. 🥺

**Reply by xp5-org** [reply]
(work in progress almost done) : https://xp5.org/index.php/86box_audio-in  


ill put in a pull request once i have the notes sorted out and can verify behavior around the mixer controls & recording gain.
